pin.go

```diff
@@ -251,3 +252,63 @@ func performPinChange(tpm *tpm2.TPMContext, public *tpm2.NVPublic, authPolicies
 
 	return nil
 }
+
+// ChangePIN changes the PIN for the key data file at the specified path. The existing PIN must be supplied via the oldPIN argument.
+// Setting newPIN to an empty string will clear the PIN and set a hint on the key data file that no PIN is set.
+//
+// If the TPM's dictionary attack logic has been triggered, a ErrTPMLockout error will be returned.
+//
+// If the file at the specified path cannot be opened, then a wrapped *os.PathError error will be returned.
+//
+// If the supplied key data file fails validation checks, an InvalidKeyFileError error will be returned.
+//
+// If oldPIN is incorrect, then a ErrPINFail error will be returned and the TPM's dictionary attack counter will be incremented.
+func ChangePIN(tpm *TPMConnection, path string, oldPIN, newPIN string) error {
+	// Check if the TPM is in lockout mode
```

Comment:
it seems we have other places that do this check? worth a helper, maybe not yet?

Reply:
I've added this check here for now as a stop-gap - I think I might remove both checks in a follow up in favour of just detecting the lockout error returned from the TPM instead.